### PR TITLE
Add metrics exporter

### PR DESCRIPTION
**Pull request by vishweshbankwar**
This is a first draft for metric exporter and not intended for release yet.

Open Items:
1. Common Schema mapping
2. Metric exporter related options e.g MetricExportInterval
3. Tests

If the draft looks good, will proceed further with open items.

### CODE REVIEW

**Review by rajkumar-rangaraj — COMMENTED** [review]
This draft looks good.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorConverter.cs [hunk 1]
```diff
@@ -58,6 +61,38 @@ internal static List<TelemetryItem> Convert(Batch<Activity> batchActivity, strin
...
+                        {
+                            BaseType = Telemetry_Base_Type_Mapping[TelemetryType.Metric],
+                            BaseData = TelemetryPartB.GetMetricData(metric, ref metricPoint)
+                        };
+                        telemetryItems.Add(telemetryItem);
+                    }
+                }
+                else
+                {
+                    // log not supported
```
Comment (reyang) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorConverter.cs:89]: Do we convert long to double? (doesn't have to address in this PR)

**Review by reyang — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/TelemetryItem.cs [hunk 2]
```diff
@@ -66,6 +68,15 @@ public TelemetryItem (LogRecord logRecord)
             this.Tags[ContextTagKeys.AiInternalSdkVersion.ToString()] = SdkVersionUtils.SdkVersion;
         }
 
+        public TelemetryItem(ref MetricPoint metricPoint)
```
Comment (cijothomas) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/TelemetryItem.cs:71]: nit: this only needs EndTime from metricpoint. So you can just pass that alone.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorConverter.cs [hunk 3]
```diff
@@ -58,6 +61,38 @@ internal static List<TelemetryItem> Convert(Batch<Activity> batchActivity, strin
...
         }
 
+        internal static List<TelemetryItem> Convert(Batch<Metric> batch, string roleName, string roleInstance, string instrumentationKey)
+        {
+            List<TelemetryItem> telemetryItems = new List<TelemetryItem>();
+            TelemetryItem telemetryItem;
+
+            foreach (var metric in batch)
+            {
+                if (metric.MetricType == MetricType.DoubleSum || metric.MetricType == MetricType.DoubleGauge)
```
Comment (cijothomas) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorConverter.cs:71]: could we not export Histograms? (ofcourse the sum/count part only)
https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Generated/Models/MetricDataPoint.cs#L37-L45

Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorConverter.cs:71]: Should we add it? We will not be able to say we support histograms :)
I will also bring this up with other language exporters to collect their feedback as well.

Comment (cijothomas) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorConverter.cs:71]: We support part of it.. So something like "histograms gets exported without the buckets" or something.
(Not blocking this PR, just something to keep in mind as this evolves).

**Pull request merged**